### PR TITLE
Improve the performance to build the SQL CLI image from 6 min to 1

### PR DESCRIPTION
**Pull request by tatiana**
## Description

As part of the SQL CLI integration to the Astro CLI, we were building a container and all its dependencies when the user ran a flow (SQL)  command for the first time.

## 🎟 Issue(s)

Related https://github.com/astronomer/astro-sdk/issues/1167

## 🧪 Functional Testing

```
docker images prune -a
astro flow --help
```

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [x] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: sql/flow.go
Comment (neel-astro) [inline sql/flow.go:83]: nit: having `astroSQLCliConfigURL` constant defined in this file itself instead of `sql/flow.go` might be better for code readability, wdyt?

File: sql/flow_config.go
Comment (neel-astro) [inline sql/flow_config.go:22]:
```suggestion
	astroSQLCLIProjectURL = "https://pypi.org/pypi/astro-sql-cli/json"
	astroSQLCLIConfigURL  = "https://raw.githubusercontent.com/astronomer/astro-sdk/astro-cli/sql-cli/config/astro-cli.json"
```
nitpick

File: sql/flow_config.go
Comment (neel-astro) [inline sql/flow_config.go:36]:
```suggestion
		return "", fmt.Errorf("error creating HTTP request %w", err)
```
nit: slightly simpler way to do the same thing, will leave it up to you on whichever way you want to go with it

File: sql/flow_config.go
Comment (neel-astro) [inline sql/flow_config.go:41]:
```suggestion
		return "", fmt.Errorf("error getting latest release version for project url %s,  %w", projectURL, err)
```

File: sql/flow_config.go
Comment (neel-astro) [inline sql/flow_config.go:49]:
```suggestion
		return "", fmt.Errorf("error parsing response for project version %w", err)
```

**Review by neel-astro — COMMENTED** [review]
Might want to sign up in Circle CI via your GitHub account, so that the Circle CI workflow could be executed for this PR, will approve once all the check passes on Circle CI.
The code looks fine overall, left minor nitpicks if you want to address them.

**Review by kushalmalani — APPROVED** [review]

**Pull request merged**
